File: pf/0 introduction/Somme.java
import java.util.stream.*;

public class Somme {
    public static void main(String args[]) {
        // System.out.println(IntStream.range(1,10).sum());
        // System.out.println(IntStream.range(1,10).map(i -> i * 2).sum());
        // System.out.println(IntStream.range(1,10).map(i -> i * i).sum());

        // Comparer les temps d’exécution entre les deux lignes
        // suivantes :
        // System.out.println(IntStream.range(1,2000000000).map(i -> i * i).sum());
        System.out.println(IntStream.range(1,2000000000).parallel().map(i -> i * i).sum());
    }
}

File: pf/0 introduction/somme.hs
somme n  = sum [1..n]
somme' n = sum (enumFromTo 1 n)

sommePairs n   = sum [2,4..2*n]
sommePairs' n  = sum (enumFromThenTo 2 4 (2*n))
sommePairs'' n = sum (map (* 2) [1..n])

sommeCarres n = sum (map (^ 2) [1..n])
